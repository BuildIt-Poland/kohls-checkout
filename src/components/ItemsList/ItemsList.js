import React from 'react';
import styled from 'styled-components';

import { spacingMedium } from '../../styles/designTokens';
import Item from '../Item/Item';

const Wrapper = styled.div`
  margin: ${spacingMedium} 0;
`;

function ItemsList({ items, handleClickUp, handleClickDown }) {
  return (
    <Wrapper>
      {items.map((item, index) => (
        <Item
          key={index}
          item={item}
          editableQuantity
          handleClickUp={() => handleClickUp(index)}
          handleClickDown={() => handleClickDown(index)}
        />
      ))}
    </Wrapper>
  );
}

export default ItemsList;
