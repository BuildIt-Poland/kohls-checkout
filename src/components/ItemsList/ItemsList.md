```js
import { connect } from 'react-redux';

function mapStateToProps(state) {
  return {
    items: state.cart.items
  };
}

const CartItemsList = connect(mapStateToProps)(ItemsList);

<CartItemsList />;
```
