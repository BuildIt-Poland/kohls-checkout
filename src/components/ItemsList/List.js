import styled from 'styled-components';

import Content from '../Content';

const List = styled(Content)`
  list-style: none;
`;

export default List;
