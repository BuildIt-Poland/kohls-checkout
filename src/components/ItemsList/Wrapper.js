import styled from 'styled-components';

import { colorLightGrey, borderWidthThin, spacingLarge } from '../../styles/designTokens';

const Wrapper = styled.section`
  margin-top: ${spacingLarge};
  border-top: ${borderWidthThin} solid ${colorLightGrey};
  padding-top: ${spacingLarge};
`;

export default Wrapper;
