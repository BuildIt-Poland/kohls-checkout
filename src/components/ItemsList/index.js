import { connect } from 'react-redux';

import ItemsList from './ItemsList';
import { updateQuantity } from '../../actions/quantityActions';

function mapStateToProps(state) {
  return {
    items: state.cart.items
  };
}

function mapDispatchToProps(dispatch) {
  return {
    handleUpdate: (cartItemId, quantity) => dispatch(updateQuantity(cartItemId, quantity))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ItemsList);
