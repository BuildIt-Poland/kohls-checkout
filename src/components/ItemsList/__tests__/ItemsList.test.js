import React from 'react';
import { create } from 'react-test-renderer';

import ItemsList from '../ItemsList';

describe('COMPONENT - ItemsList', () => {
  it('renders ItemsList component', () => {
    const component = create(<ItemsList items={[]} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
