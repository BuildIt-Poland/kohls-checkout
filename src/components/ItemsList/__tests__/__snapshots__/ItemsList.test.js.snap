// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - ItemsList renders correct number of items 1`] = `
.c0 {
  margin-top: 3rem;
  border-top: 1px solid #b0b0b0;
  padding-top: 3rem;
}

.c1 {
  max-width: 100rem;
  padding: 0 2rem;
  margin: 0 auto;
  list-style: none;
}

<section
  className="c0"
>
  <ul
    className="c1"
  >
    <Item
      item={
        Object {
          "id": "64a673c2-59e3-11e9-8647-d663bd873d93",
          "imgUrl": "https://mosaic03.ztat.net/vgs/media/catalog_hd/LE/42/2T/01/0K/11/LE422T010-K11@8.1.jpg",
          "name": "RIDER JACKET - jeans jacket - dark blue",
          "price": Object {
            "discount": 12500,
            "regular": 24099,
          },
          "quantity": 3,
          "variants": Object {
            "color": "Dark Blue",
            "size": "46 Medium",
          },
        }
      }
    />
    <Item
      item={
        Object {
          "id": "71380d3a-59e3-11e9-8647-d663bd873d93",
          "imgUrl": "https://mosaic03.ztat.net/vgs/media/catalog_hd/C7/04/1F/06/LL/11/C7041F06L-L11@9.jpg",
          "name": "WOMAN JACKET FIX HOOD - outdoor - mint",
          "price": Object {
            "regular": 44000,
          },
          "quantity": 4,
          "variants": Object {
            "color": "Mint",
            "size": "42 Short",
          },
        }
      }
    />
  </ul>
</section>
`;
